lua/iswap/internal.lua

```diff
@@ -7,13 +7,13 @@ local ft_to_lang = require('nvim-treesitter.parsers').ft_to_lang
 local M = {}
 
 --
-function M.find(winnr)
-  local bufnr = vim.fn.winbufnr(winnr)
-  local cursor = vim.api.nvim_win_get_cursor(vim.fn.win_getid(winnr))
+function M.find(winid)
+  local bufnr = vim.api.nvim_win_get_buf(winid)
+  local cursor = vim.api.nvim_win_get_cursor(winid)
   local cursor_range = { cursor[1] - 1, cursor[2] }
   local row = cursor_range[1]
   local root = ts_utils.get_root_for_position(unpack(cursor_range))
-  local ft = vim.api.nvim_buf_get_option(bufnr, 'filetype')
+  local ft = vim.bo[bufnr].filetype
```

Comment:
Are you sure this is better practice than using the API function `nvim_buf_get_option`? I have never seen indexing into `vim.bo` before but you're more knowledgeable about this than I am.

Reply:
FYI: https://github.com/nanotee/nvim-lua-guide#using-meta-accessors
`:h vim.bo`
https://github.com/neovim/neovim/blob/fccc4ed22d58ca4005e4acb144d6668ba094b8c7/runtime/lua/vim/_meta.lua#L77-L105